integration/tests/neon_evm/test_compute_units.py

```diff
@@ -0,0 +1,563 @@
+    deterministic_operator_keypair: Keypair,
+    deterministic_key_pairs: dict[Literal["sender_with_tokens", "user"], list[Keypair]],
+) -> Caller:
+    mark: pytest.Mark = request.node.get_closest_marker("deterministic_user_index")
+    index = mark.args[0]
+    key = deterministic_key_pairs["user"][index]
+    return evm_loader.make_new_user(deterministic_operator_keypair, key=key)
+
+
+@pytest.fixture
+def deterministic_holder_acc(
+    request: pytest.FixtureRequest, deterministic_operator_keypair: Keypair, evm_loader: EvmLoader
+) -> Pubkey:
+    mark: pytest.Mark = request.node.get_closest_marker("deterministic_holder_acc_seed")
+    seed = f"{mark.args[0]}_seed"
+    return evm_loader.create_holder(signer=deterministic_operator_keypair, seed=seed)
+
+
+def allure_attach_accounts_data(
+    resp: GetTransactionResp,
+    evm_loader: EvmLoader,
+    title: str = "Used accounts data",
+):
+    accounts_data = {}
+
+    for pubkey in resp.value.transaction.transaction.message.account_keys:
+        info = evm_loader.get_account_info(pubkey).value
+
+        if info:
+            accounts_data[str(pubkey)] = str(info.data)
+
+    allure.attach(
+        body=json.dumps(obj=accounts_data, indent=2),
+        name=title,
+        attachment_type=allure.attachment_type.JSON,
+    )
+
+
+def execute_transaction_steps_from_instruction_and_validate_cu(
+    evm_loader: EvmLoader,
+    operator: Keypair,
+    treasury,
+    storage_account,
+    instruction: SignedTransaction,
+    additional_accounts,
+    cu_expected_list: list[int],
+    cu_delta_allowed: int,
+    signer: Keypair = None,
+    compute_unit_price=None,
+    chain_id: int | None = None,
+):
+    chain_id = chain_id or evm_loader.chain_id
+
+    signer = operator if signer is None else signer
+    operator_balance_pubkey = evm_loader.get_operator_balance_pubkey(operator, chain_id)
+    index = 0
+
+    for i, cu_expected in enumerate(cu_expected_list):
+        receipt = evm_loader.send_transaction_step_from_instruction(
+            operator,
+            operator_balance_pubkey,
+            treasury,
+            storage_account,
+            instruction,
+            additional_accounts,
+            EVM_STEPS,
+            signer,
+            compute_unit_price=compute_unit_price,
+            index=index,
+        )
+        index += 1
+        if receipt.value.transaction.meta.err:
+            raise AssertionError(f"Transaction failed with error: {receipt.value.transaction.meta.err}")
+        for log in receipt.value.transaction.meta.log_messages:
+            if "exit_status" in log:
+                break
+            if "ExitError" in log:
+                raise AssertionError(f"EVM Return error in logs: {receipt}")
+
+        allure_attach_accounts_data(resp=receipt, evm_loader=evm_loader, title=f"Used accounts data {i}")
+
+        cu_consumed = receipt.value.transaction.meta.compute_units_consumed
+        assert (cu_consumed - cu_expected) <= cu_delta_allowed
```

Comment:
I think it will be good to check that transactions is finished , there are no more iterations